sklearn/manifold/isomap.py

```diff
@@ -109,13 +202,29 @@ def _fit_transform(self, X):
         kng = kneighbors_graph(self.nbrs_, self.n_neighbors,
                                mode='distance')
 
-        self.dist_matrix_ = graph_shortest_path(kng,
-                                                method=self.path_method,
-                                                directed=False)
-        G = self.dist_matrix_ ** 2
-        G *= -0.5
+        self.dist_matrix_ = shortest_path(kng,
+                                          method='D',
+                                          directed=False,
+                                          indices=landmarks).T
 
-        self.embedding_ = self.kernel_pca_.fit_transform(G)
+        self.dist_matrix_[np.where(np.isinf(self.dist_matrix_))] = 0
```

Comment:
Without this, I'm getting this error in `sklearn/tests/test_commons.py`:
```shell
ERROR: sklearn.tests.test_common.test_non_meta_estimators('Isomap', <class 'sklearn.manifold.isomap.Isomap'>)
# ...
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
```

My guess was that scipy's functions didn't replace inf with zeros at the end of the methods, as sklearn's [version does](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/graph_shortest_path.pyx#L152). I didn't want to modify this in `shortest_path` because that's a complete logic change.